test/evergreen.yml

```diff
@@ -2236,7 +2236,7 @@ tasks:
           script: |
             set -o errexit
             set -o verbose
-            for i in {1..10}; do ${python_binary|python3} split_stress.py; done
+            for i in {1..10}; do ${test_env_vars|} ${python_binary|python3} split_stress.py; done
```

Comment:
Just for my understanding, the `python_binary` part was already there. Below you add a lot of lines to set `python_binary`. Where was it coming from before?

Reply:
Before, `python_binary` was set only for Windows.  So, before this change, it was defaulting to `python3` on Unix, which would get the toolchain python only if the PATH was set correctly.